tests/test_update_runnablejobs.py

```diff
@@ -0,0 +1,175 @@
+                                                platform_option=platform_option)
+                    results.append({'build_system_type': build_system_type,
+                                    'platform_option': platform_option,
+                                    'ref_data_name': ref_data_name,
+                                    'build_platform': build_platform})
+            else:
+                for build_platform in ["linux64", "android-4-3-armv7-api15"]:
+                    temp = "desktop-test-{build_platform}/" \
+                           "{platform_option}-test-for-{build_platform}"
+                    ref_data_name = temp.format(build_platform=build_platform,
+                                                platform_option=platform_option)
+                    results.append({'build_system_type': build_system_type,
+                                    'platform_option': platform_option,
+                                    'job_type_name': ref_data_name,
+                                    'ref_data_name': ref_data_name,
+                                    'build_platform': build_platform})
+
+    return {"results": results}
+
+
+class TestUpdateRunnableapi(unittest.TestCase):
+    """This class is used to test the update_runnableapi function"""
+
+    MOCK_DATA = data_generator()
+    MOCK_PUSH = {"taskId": 123456, "expires": "2017-07-04T22:13:23"}
+
+    def setUp(self):
+        """Setting up values that will be used in every test."""
+
+        self.expected = self.MOCK_DATA
+
+        # we need to create two timetamp to mock the new one and old one
+        today = datetime.date.today().strftime('%Y-%m-%d')
+        time_tuple = datetime.datetime.strptime(today, "%Y-%m-%d").timetuple()
+        old_time = (datetime.date.today() -
+                    datetime.timedelta(days=1)).strftime('%Y-%m-%d')
+        old_time_tuple = datetime.datetime.strptime(old_time, "%Y-%m-%d").timetuple()
+        self.new_timestamp = time.mktime(time_tuple)
+        self.old_timestamp = time.mktime(old_time_tuple)
+
+        # rename the runnablejob.json and preseed.json file untill the end of test
+        try:
+            if os.path.isfile(ORI_RUNNABLE_PATH):
+                os.rename(ORI_RUNNABLE_PATH, TMP_RUNNABLE_FILENAME)
+            shutil.copy2(ORI_PRESEED_PATH, TMP_PRESEED_FILENAME)
+
+        except Exception as e:
+            logger.warning(e)
+
+    def tearDown(self):
+        """Clean up after every test."""
+        try:
+            if os.path.isfile(ORI_RUNNABLE_PATH):
+                os.remove(ORI_RUNNABLE_PATH)
+                os.rename(TMP_RUNNABLE_FILENAME, ORI_RUNNABLE_PATH)
+
+        except Exception as e:
+            logger.warning(e)
+
+        try:
+            if os.path.isfile(ORI_PRESEED_PATH):
+                os.remove(ORI_PRESEED_PATH)
+                os.rename(TMP_PRESEED_FILENAME, ORI_PRESEED_PATH)
+                os.rename(TMP_PRESEED_FILENAME, ORI_PRESEED_PATH)
```

Comment:
oh, the line 125 is duplicate with line 124. I will remove it!

The logic in here is, we store the original preseed.json as tmp_preseed.json and we use its copy to test the add_new_jobs_into_pressed(). After we finished the test, we should remove the copy and restore the tmp_preseed.json to the preseed.json.